contracts/ProposalsContract.sol

```diff
@@ -0,0 +1,183 @@
+
+	modifier onlyMultisigAddress() {
+		require(msg.sender==multisigAddress);
+		_;
+	}
+
+	/**
+	 * @notice _bridge SHOULD CALL transferOwnership() to THIS contract!
+	 * @param _bridge – address of the bridge contract (that we will control)
+	 * @param _token – address of the main DAO token
+	 * @param _multisigAddress – address of the mulitisig contract (that controls us)
+	 */
+	constructor(IBridgeContract _bridge, PreserveBalancesOnTransferToken _token, address _multisigAddress) public {
+		multisigAddress = _multisigAddress;
+		bridge = _bridge;
+		token = _token;
+	}
+
+	/**
+	 * @dev Propose the 'exitStake' parameter change
+	 * @notice This function creates voting
+	 * @param _exitStake – value of exitStake param
+	 */
+	function setExitStake(uint256 _exitStake) public onlyMultisigAddress {
+		Voting memory v;
+		v.votingType = VotingType.SetExitStake;
+		v.param = _exitStake;
+		v.eventId = token.startNewEvent();
+		v.pro = 0;
+		v.versus = 0;
+		v.totalSupplyAtEvent = token.totalSupply();
+		votings.push(v);
+	
+		emit VotingStarted("setExitStake", _exitStake, v.totalSupplyAtEvent, v.eventId, msg.sender);
+	}
+
+	/**
+	 * @dev Propose the 'epochLength' parameter change
+	 * @notice This function creates voting
+	 * @param _epochLength – value of epochLength param
+	 */
+	function setEpochLength(uint _epochLength) public onlyMultisigAddress {
+		Voting memory v;
+		v.votingType = VotingType.SetEpochLength;
+		v.param = _epochLength;
+		v.eventId = token.startNewEvent();
+		v.pro = 0;
+		v.versus = 0;
+		v.totalSupplyAtEvent = token.totalSupply();
+		votings.push(v);
+		
+		emit VotingStarted("setEpochLength", _epochLength, v.totalSupplyAtEvent, v.eventId, msg.sender);
+	}
+
+	/**
+	 * @dev Get ALL voting count 
+	 * @notice This function can be called by anyone
+	 * @return Voting count
+	 */
+	function getVotingsCount()public view returns(uint){
+		return votings.length;
+	}
+
+	/**
+	* @dev Get voting data
+	 * @notice This function can be called by anyone
+	 * @param _votingIndex – voting number
+	 * @return votingType – what is this voting for
+	 * @return paramValue – what is param amount
+	 * @return versus – sum of voters token amount, that voted no
+	 * @return isFinished – is Quorum reached
+	 * @return isResultYes – is voted yes >= 80%
+	 */
+	function getVotingStats(uint _votingIndex) public view returns(VotingType votingType, uint paramValue, uint pro, uint versus, bool isFinished, bool isResultYes) {
+		require(_votingIndex<votings.length);
+		votingType = votings[_votingIndex].votingType;
+		paramValue = votings[_votingIndex].param;		
+		pro = votings[_votingIndex].pro;
+		versus = votings[_votingIndex].versus;
+		isFinished = _isVotingFinished(_votingIndex);
+		isResultYes = _isVotingResultYes(_votingIndex);
+	}
+
+	/**
+	 * @dev Is selected voting finished?
+	 * @param _votingIndex – voting number
+	 * @return is quorum reched or not
+	 */
+	function _isVotingFinished(uint _votingIndex) internal returns(bool isFin) {
```

Comment:
so for a proposal to be applied, a certain amount of YES votes needed? If so, I think this is wrong cause we want to go with reactive approach for Proposals (see https://docs.google.com/document/d/1fC7YFgJB2QQbqbyJuU1AVqleLQPf6hEp0qNgj9-L_dM/edit#heading=h.o16xz6ryknt6). Voting to be used only for Confidence Vote (which is basically changing the owner of the bridge)